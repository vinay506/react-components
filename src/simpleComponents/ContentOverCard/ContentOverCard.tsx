import React from 'react'
import styled from 'styled-components';
function ContentOverCard(props: any) {
    /**
     * backgroundColor: defaults to light blue, if required pass the color code or color name to override
     */
    const { backgroundColor, overRideStyles } = props;
    const propStyles = {
        backgroundColor: backgroundColor, ...overRideStyles
    }
    return (
        <React.Fragment>
            <ContentOverContainer id="uniqueContentOverride" style={propStyles}>
                {props.children}
            </ContentOverContainer>
        </React.Fragment>
    )
}

const ContentOverContainer = styled.div`
background-color: #dae8ed;
display: flex;
min-height: 250px;
justify-content: center;
`
export default ContentOverCard
