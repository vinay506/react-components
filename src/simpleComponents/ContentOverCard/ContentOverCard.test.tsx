import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import ContentOverCard from './ContentOverCard';
import 'jest-styled-components'
describe('Content Over Card', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles = {
        backgroundColor: '#dae8ed',
        display: 'flex',
        minHeight: '250px',
        justifyContent: 'center'
    };
    beforeEach(() => {
        propInfo = {
            backgroundColor: 'green'
        };
        wrapper = shallow(<ContentOverCard {...propInfo}><h1 id="innerText">Sample</h1></ContentOverCard>);
    });
    test('Content Over Card Snapshot', () => {
        const instance = renderer.create(<ContentOverCard {...propInfo}><h1>Sample</h1></ContentOverCard>);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Content Over Card Element check', () => {
        const elementLength = wrapper.find('#uniqueContentOverride').length;
        expect(elementLength).toEqual(1);
    })

    // Default style Validations
    test('Content Over Card Default Background Validation', () => {
        const instance = renderer.create(<ContentOverCard {...propInfo}><h1>Sample</h1></ContentOverCard>);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.backgroundColor);
    })

    test('Content Over Card Default display Validation', () => {
        const instance = renderer.create(<ContentOverCard {...propInfo}><h1>Sample</h1></ContentOverCard>);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.display);
    })

    test('Content Over Card Default Min Height Validation', () => {
        const instance = renderer.create(<ContentOverCard {...propInfo}><h1>Sample</h1></ContentOverCard>);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.minHeight);
    })

    test('Content Over Card Default Justify Content Validation', () => {
        const instance = renderer.create(<ContentOverCard {...propInfo}><h1>Sample</h1></ContentOverCard>);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.justifyContent);
    })

    // Props Validation
    test('Content Over Card Inner Elements check', () => {
        const children = renderer.create(<ContentOverCard {...propInfo}><h1>Sample</h1></ContentOverCard>).toJSON()?.children;
        if (children && children.length > 0) {
            expect(children[0]?.type).toEqual('h1');
            expect(children[0].children[0]).toEqual('Sample');
        }
    })

})