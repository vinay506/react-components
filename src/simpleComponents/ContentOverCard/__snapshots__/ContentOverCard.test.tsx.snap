// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Over Card Content Over Card Snapshot 1`] = `
.c0 {
  background-color: #dae8ed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 250px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
  id="uniqueContentOverride"
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  <h1>
    Sample
  </h1>
</div>
`;
